Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -222,11 +224,14 @@ IRelocator relocator
       documents.ToDictionary(k => k.Key, v => (IDocumentEntry)v.Value);
 
     private record DocumentEntry(int? Version, Task<DafnyDocument> ResolvedDocument,
-      Task<DafnyDocument> VerifiedDocument,
+      Task<DafnyDocument> FullyVerifiedDocument,
       CancellationTokenSource CancellationSource) : IDocumentEntry {
       public void CancelPendingUpdates() {
         CancellationSource.Cancel();
       }
+
+      public Task<DafnyDocument> MostVerifiedDocument =>
+        FullyVerifiedDocument.IsCompletedSuccessfully ? FullyVerifiedDocument : ResolvedDocument;
```

Comment:
I'm not sure about these names: resolution isn't verification.  Not sure I can think of something better.  Maybe `checked`?  We do resolution and typechecking, then verification; a verified document is "fully checked", whereas a typechecked one is only "partially checked".